spinn_machine/tags/iptag.py

```diff
@@ -31,8 +32,10 @@ class IPTag(AbstractTag):
 
     # pylint: disable=too-many-arguments
     def __init__(
-            self, board_address, destination_x, destination_y, tag, ip_address,
-            port=None, strip_sdp=False, traffic_identifier="DEFAULT"):
+            self, board_address: Optional[str], destination_x: int,
+            destination_y: int, tag: int, ip_address: str,
+            port: Optional[int] = None, strip_sdp: bool = False,
+            traffic_identifier: str = "DEFAULT"):
```

Comment:
Worth checking if these optionals are ever used optionally; it may clear up some other issues if they are not optional.  As there is an IPTagResource in PACMAN that is used to specify one of these is wanted, during allocation it could be that the board_address is fixed.  The port is less clear, though I think a port is going to be needed, so maybe again this is fixed here?

Reply:
board_address does not need to be Optional

Reply:
port looks like it does need to be Optional coming from places like EthernetControlPopulation

Reply:
Though the port passed in is optional, it should resolve before the tag is set.  That said, I suppose a tag can have "no port" if it is going to be set by sending a message later (this sometimes happens).  I think it still has to have some sort of port though, so perhaps the default is 0 or something, rather than None...

Reply:
Note that the resolution of the port is because the port parameter (local_port I think) is then given to a connection which will then create an "any port" connection.  This will be assigned by the OS to something real though and then the local_port property of the connection has the real port number.  This can be used regardless of it is really going to be set later if that helps...

Reply:
After farther testing I was able to make both board_address and port none Optional